Sub/Src/ComputerVision/libs/pose_calculation.py

```diff
@@ -115,42 +118,77 @@ def set_coordinates(self, detect_list, detection, x, y, w, h):
+            self.gate_right_points = np.array([[self.max, self.quarter_min, self.center], #5, -1.25, 0
+                                               [self.max, self.center, self.center], #5, 0, 0
+                                               [self.max, self.quarter_max, self.center], #5, 1.25, 0
+                                               [self.max, self.mid_max, self.center]]) #5, 2.5, 0
 
-            self.gate_left_points = np.array([[self.min, self.quarter_min, self.center],
-                                              [self.min, self.center, self.center],
-                                              [self.min, self.quarter_max, self.center],
-                                              [self.min, self.mid_max, self.center]])
+            self.gate_left_points = np.array([[self.min, self.quarter_min, self.center], #-5, -1.25, 0
+                                              [self.min, self.center, self.center], #-5, 0, 0
+                                              [self.min, self.quarter_max, self.center], #5, 1.25, 0
+                                              [self.min, self.mid_max, self.center]]) #5, 2.5, 0
 
             for second_det in detect_list:
-                if (second_det[0] == b'Gate Arm'):
-                    self.second_x_coordinate, self.second_y_coordinate, self.second_width, self.second_height = second_det[2][0], second_det[2][1], second_det[2][2], second_det[2][3]
-                    difference = self.second_x_coordinate - self.x_coordinate
-                    if(difference > 0):
-                        self.three_dim_points = np.concatenate((self.gate_top_points, self.gate_right_points), axis = 0)
-                    else:
-                        self.three_dim_points = np.concatenate((self.gate_top_points, self.gate_left_points), axis = 0)
-
-                    self.two_dim_points = np.array([[self.x_coordinate, self.y_coordinate + (self.second_y_coordinate - (0.5 * self.second_height))],
-                                                    [self.x_coordinate - (0.5 * self.width), self.y_coordinate],
-                                                    [self.x_coordinate - (0.25 * self.width), self.y_coordinate],
-                                                    [self.x_coordinate, self.y_coordinate],
-                                                    [self.x_coordinate + (0.25 * self.width), self.y_coordinate],
-                                                    [self.x_coordinate + (0.5 * self.width), self.y_coordinate],
-                                                    [self.second_x_coordinate, self.second_y_coordinate - (0.25 * self.second_height)],
+                self.second_x_coordinate, self.second_y_coordinate, self.second_width, self.second_height = second_det[2][0], second_det[2][1], second_det[2][2], second_det[2][3]
+                self.difference = self.second_x_coordinate - self.x_coordinate
+
+                if(second_det[0] == b'Gate Top'):
```

Comment:
Since the values are specific for each object type, make these calculations in individual functions. Also please comment more on what is happening.

Reply:
A lot of the code looks repetative, so I think it can be modularized into functions.